### PR TITLE
Prevent use ObjectGraphType in QueryArgument

### PR DESCRIPTION
**Pull request by sungam3r**
Relates to #1436 

By the way, two tests were written incorrectly. Fixed.

### CODE REVIEW

**Comment by sungam3r** [comment]
@joemcbride If you don't mind I would like to merge these changes ASAP.

File: src/GraphQL/Types/QueryArgument.cs [hunk 1]
```diff
@@ -37,8 +40,35 @@ public QueryArgument(Type type)
...
+        {
+            if (type?.IsInputType() == false)
+                throw Create(nameof(Type), type);
+
+            return type;
+        }
+
+        private IGraphType CheckResolvedType(IGraphType type)
+        {
+            if (!(type.GetNamedType() is GraphQLTypeReference) && type?.IsInputType() == false)
```
Comment (joemcbride) [inline src/GraphQL/Types/QueryArgument.cs]: Just noticed, I think missing a `type?` here on the `type.GetNamedType()`.

Maybe add a test with null for the `ResolvedType` property?

Comment (sungam3r) [reply src/GraphQL/Types/QueryArgument.cs]: I already raised a finger over the merge button :)

```c#
  public static IGraphType GetNamedType(this IGraphType type)
        {
            return type switch
            {
                NonNullGraphType nonNull => GetNamedType(nonNull.ResolvedType),
                ListGraphType list => GetNamedType(list.ResolvedType),
                _ => type
            };
        }
```

Comment (sungam3r) [reply src/GraphQL/Types/QueryArgument.cs]: I would say that on the contrary we can remove `?` when used before `IsInputType` or `IsOutputType` (`IsOutputType` requires small fix).

Comment (sungam3r) [reply src/GraphQL/Types/QueryArgument.cs]: image

Nope, nothing to fix.

Comment (joemcbride) [reply src/GraphQL/Types/QueryArgument.cs]: I would still like to see a test that sets `ResolvedType` to null to confirm the behavior.

Comment (sungam3r) [reply src/GraphQL/Types/QueryArgument.cs]: Ok. I was wrong about `IsInputType`, `bool?` vs `bool`.

Comment (sungam3r) [reply src/GraphQL/Types/QueryArgument.cs]: Done
